resources/server_install.rb

```diff
@@ -49,21 +49,59 @@
 
   log 'Enable and start MariaDB service' do
     notifies :enable, "service[#{platform_service_name}]", :immediately
-    notifies :start, "service[#{platform_service_name}]", :immediately
+    notifies :stop, "service[#{platform_service_name}]", :immediately
+    notifies :run, 'execute[apply-mariadb-root-password]', :immediately
   end
 
-  mariadb_root_password = new_resource.password == 'generate' || new_resource.password.nil? ? secure_random : new_resource.password
+  # here we want to generate a new password if: 1- the user passed 'generate' to the password argument
+  #                                             2- the user did not pass anything to the password argument OR
+  #                                                the user did not define node['mariadb']['server_root_password'] attribute
+  mariadb_root_password = (new_resource.password == 'generate' || new_resource.password.nil?) ? secure_random : new_resource.password
 
-  # Generate a ramdom password or set the a password defined with node['mariadb']['password']['root'].
+  # Generate a random password or set a password defined with node['mariadb']['server_root_password'].
   # The password is set or change at each run. It is good for security if you choose to set a random password and
   # allow you to change the root password if needed.
-  bash 'generate-mariadb-root-password' do
+  file 'generate-mariadb-root-password' do
```

Comment:
Do we really want to be putting the password in plain text on the server?

Reply:
I see your point. However, the file is only accessible by root and honestly if the machine is somehow owned, the attacker could always access the db server anyway.

In the end the created file can be removed, but (to my knowledge) it has to exist, keep in mind the `--init-file=` argument needed to pass to mysqld to reset the root password.

Of course other solutions are welcome.